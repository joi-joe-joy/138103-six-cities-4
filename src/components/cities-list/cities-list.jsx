import React from "react";
import classnames from "classnames";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer.js";
import pt from 'prop-types';

const CitiesList = (props) => {
  const {onLocationClick, currentCity, citiesList} = props;

  return (
    <div className="tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {citiesList.map((city) => (
            <li key={city} className="locations__item">
              <a className={classnames(`locations__item-link tabs__item`, {
                'tabs__item--active': currentCity === city
              })} href="#" onClick={(e) => {
                e.preventDefault();
                onLocationClick(city);
              }}>
                <span>{city}</span>
              </a>
            </li>
          ))}
        </ul>
      </section>
    </div>
  );
};

CitiesList.propTypes = {
  onLocationClick: pt.func.isRequired,
  citiesList: pt.arrayOf(pt.string).isRequired,
  currentCity: pt.string,
};

const mapStateToProps = (state) => ({
  currentCity: state.city,
  citiesList: state.citiesList
});

const mapDispatchToProps = (dispatch) => ({
  onLocationClick(city) {
    dispatch(ActionCreator.changeCity(city));
    dispatch(ActionCreator.getOffersCityList(city));
  }
});

export {CitiesList};
export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
