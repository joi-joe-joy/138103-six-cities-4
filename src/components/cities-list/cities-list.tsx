import * as React from "react";
import * as classnames from "classnames";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer/data/data";
import {getCity, getCitiesList} from "../../reducer/data/selectors";
import {City} from "../../types";

interface Props {
  onLocationClick: (city: City) => void;
  citiesList: City[];
  currentCity: City;
}

const CitiesList: React.FC<Props> = (props: Props) => {
  const {onLocationClick, currentCity, citiesList} = props;

  return (
    <div className="tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {citiesList.map((city) => (
            <li key={city.name} className="locations__item">
              <a className={classnames(`locations__item-link tabs__item`, {
                'tabs__item--active': currentCity.name === city.name
              })} href="#" onClick={(e) => {
                e.preventDefault();
                onLocationClick(city);
              }}>
                <span>{city.name}</span>
              </a>
            </li>
          ))}
        </ul>
      </section>
    </div>
  );
};

const mapStateToProps = (state) => ({
  currentCity: getCity(state),
  citiesList: getCitiesList(state)
});

const mapDispatchToProps = (dispatch) => ({
  onLocationClick(city) {
    dispatch(ActionCreator.changeCity(city));
  }
});

export {CitiesList};
export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
