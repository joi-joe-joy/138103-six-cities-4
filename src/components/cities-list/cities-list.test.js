import React from "react";
import renderer from "react-test-renderer";
import {CitiesList} from "./cities-list";

describe(`Render CitiesList correctly`, () => {
  it(`Render CitiesList correctly without currentCity`, () => {
    const tree = renderer.create(
        <CitiesList onLocationClick={()=>{}}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render CitiesList correctly with currentCity`, () => {
    const tree = renderer.create(
        <CitiesList
          currentCity={`Paris`}
          onLocationClick={()=>{}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
