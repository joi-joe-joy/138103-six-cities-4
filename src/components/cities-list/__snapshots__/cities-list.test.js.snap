// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CitiesList correctly Render CitiesList correctly with currentCity 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
        onClick={[Function]}
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
        onClick={[Function]}
      >
        <span>
          Amsterdam
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
        onClick={[Function]}
      >
        <span>
          Brussels
        </span>
      </a>
    </li>
  </ul>
</section>
`;

exports[`Render CitiesList correctly Render CitiesList correctly without currentCity 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href="#"
        onClick={[Function]}
      >
        <span>
          Paris
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
        onClick={[Function]}
      >
        <span>
          Amsterdam
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
        onClick={[Function]}
      >
        <span>
          Brussels
        </span>
      </a>
    </li>
  </ul>
</section>
`;
